assets/scripts/menus/SettingsMenu.jsx

```diff
@@ -1,17 +1,92 @@
 import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
 import { FormattedMessage } from 'react-intl'
 import Menu from './Menu'
 import LocaleDropdown from './LocaleDropdown'
+import { SETTINGS_UNITS_IMPERIAL, SETTINGS_UNITS_METRIC } from '../users/constants'
+import { updateUnits } from '../users/localization'
+import { changeLocale } from '../store/actions/locale'
+import { clearMenus } from '../store/actions/menus'
+
+export class SettingsMenu extends React.PureComponent {
+  static propTypes = {
+    units: PropTypes.number,
+    locale: PropTypes.string,
+    level: PropTypes.number,
+    changeLocale: PropTypes.func,
+    clearMenus: PropTypes.func
+  }
+
+  static defaultProps = {
+    level: 4,
+    changeLocale: () => {},
+    clearMenus: () => {}
+  }
+
+  selectLocale = (locale) => {
+    if (this.props.locale === locale) return
+
+    this.props.clearMenus()
+    this.props.changeLocale(locale)
+  }
+
+  selectMetric = () => {
+    if (this.props.units === SETTINGS_UNITS_METRIC) return
+
+    updateUnits(SETTINGS_UNITS_METRIC)
+  }
+
+  selectImperial = () => {
+    if (this.props.units === SETTINGS_UNITS_IMPERIAL) return
+
+    updateUnits(SETTINGS_UNITS_IMPERIAL)
+  }
 
-export default class SettingsMenu extends React.PureComponent {
   render () {
     return (
       <Menu onShow={this.onShow} {...this.props}>
-        <div className="form">
-          <p><FormattedMessage id="menu.language.heading" defaultMessage="Language" /></p>
-          <LocaleDropdown />
-        </div>
+        <h2 className="menu-header">
+          <FormattedMessage id="settings.units.label" defaultMessage="Units" />
+        </h2>
+        <ul className="menu-item-group">
+          <li className={`menu-item ${(this.props.units === SETTINGS_UNITS_METRIC) ? 'menu-item-selected' : ''}`} onClick={this.selectMetric}>
+            {/* &#x200E; prevents trailing parentheses from going in the wrong place in rtl languages */}
+            <FormattedMessage id="settings.units.metric" defaultMessage="Metric units (meters)" />&#x200E;
+          </li>
+          <li className={`menu-item ${(this.props.units === SETTINGS_UNITS_IMPERIAL) ? 'menu-item-selected' : ''}`} onClick={this.selectImperial}>
+            <FormattedMessage id="settings.units.imperial" defaultMessage="Imperial units (feet)" />&#x200E;
+          </li>
+        </ul>
+
+        <h2 className="menu-header">
+          <FormattedMessage id="settings.language.label" defaultMessage="Language" />
+        </h2>
+        <LocaleDropdown locale={this.props.locale} level={this.props.level} selectLocale={this.selectLocale} />
       </Menu>
     )
   }
 }
+
+function mapStateToProps (state) {
+  // The lowest level marked "true" takes priority.
+  let level = 4
+  if (state.flags.LOCALES_LEVEL_3.value) level = 3
```

Comment:
How do these locale levels work? Are certain levels hidden by default or something? Might be beneficial to have a comment here relaying that.

Reply:
Right, so level 4 is always on (this is English, the language we've always had).
Levels 1-3 were off by default. We flipped level 3 (ready to go, non English) to "on" to make those languages available to everyone. Levels 1 and 2 will not be turned permanently on but we envision we would flip those switches for certain users/environments for testing.

If this system doesn't make any sense, let me know if you think we can improve it.